protocol/client/stream.go

```diff
@@ -153,9 +153,18 @@ func (s *Stream) processLoop(ctx context.Context, messages <-chan *protocol.Mess
 	for {
 		select {
 		case msg := <-messages:
+			// TODO(adam): refactor this. For now, we mark each message as unread by default
+			// There are a few problems with putting it like this:
+			//   * Flags (and some other fields) are not in the spec so ideally
+			//     we should move them to `client`.
+			//   * It is not obvious that additional processing happens here.
+			//     It should be clear from the `Stream` caller perspective.
+			msg.Flags.Set(protocol.MessageUnread)
```

Comment:
@dshulyak in my opinion it looks even worse. Alternatively, I can create a method `Message.applyDefaults()` which would do that but the problem is that `Message` can be created in many places (`createTextMessage` and `DecodeMessage`) so `applyDefaults` would need to be called in all of them.

This further suggests that setting read/unread should happen in `client` and that `Flags` should be also in `client` using composition.